server/conn.go

```diff
@@ -2190,9 +2190,13 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 // fetchSize, the desired number of rows to be fetched each time when client uses cursor.
 func (cc *clientConn) writeChunksWithFetchSize(ctx context.Context, rs ResultSet, serverStatus uint16, fetchSize int) error {
 	fetchedRows := rs.GetFetchedRows()
+	// if fetchedRows is not enough, getting data from recordSet.
 	for len(fetchedRows) < fetchSize {
-		// if fetchedRows is not enough, getting data from recordSet.
-		req := rs.NewChunk(cc.chunkAlloc)
+		// NOTE: chunk should not be allocated from the allocator
+		// the allocator will reset every statement
+		// but it maybe stored in the result set among statements
+		// ref https://github.com/pingcap/tidb/blob/7fc6ebbda4ddf84c0ba801ca7ebb636b934168cf/server/conn_stmt.go#L233-L239
+		req := rs.NewChunk(nil)
```

Comment:
Is it better to reuse the chunk and call `req.Reset()` for every loop for reduce allocs ? 

rest LGTM .

Reply:
@xhebox What do u think?

Reply:
addressed.